### DISCUSSION TITLE
fwupd Bitbake Recipe

### DISCUSSION THREAD

**Discussion by wltu** [fwupd/fwupd#3939] [Q&A]
Hi all,

I am just wondering if there is a bitbake recipe for fwupd that is more up to date.

I know there is https://github.com/intel/intel-iot-refkit/blob/master/meta-refkit-core/recipes-extended/fwupd/fwupd_0.9.7.bb and is wondering if there is one that is in a more common meta-layer.

If not, I am thinking of upstreaming it to Yocto/OpenEmbedded

OpenBMC Discussion in: https://gerrit.openbmc-project.xyz/c/openbmc/openbmc/+/48513

Best,

Willy Tu

**Comment by superm1** [comment]
>I know there is https://github.com/intel/intel-iot-refkit/blob/master/meta-refkit-core/recipes-extended/fwupd/fwupd_0.9.7.bb and is wondering if there is one that is in a more common meta-layer.

That looks really old.

>If not, I am thinking of upstreaming it to Yocto/OpenEmbedded

If you would like to, we accept contributions like this in `contrib/` too.  You can add a new CI target to build it using your recipe and then it can stay healthier and up to date here.

**Reply by wltu** [reply]
Thanks for the answer.

For `contrib/`, do you mean like adding a new Dockerfile to enable a new CI to maintain? I am wondering how that would work if I want to make sure that the bitbake recipe will work. Do you have some example that is existing with bitbake + the CI?

**Reply by superm1** [reply]
Yeah exactly, adding a new docker file for yocto builds and running it for a CI target.

Sorry I've got no experience with using bitbake not yocto though.

 I guess I would suggest you find out if there's a docker file that can run yacto first. If there is then try to develop a recipe in that.

**Reply by wltu** [reply]
ok, sounds good. Thanks. I'll try to see if that is possible.

Just a side note, there seem to be a newer WIP version in OpenEmbeded https://gitlab.com/rossburton/meta-openembedded/-/tree/fwupd/meta-oe/recipes-bsp/fwupd

**Comment by hughsie** [comment]
Some quick comments on the deps (sorry, it's getting late here):

	+DEPENDS = " \
	...
	+  polkit \

You really don't need or want this on a BMC.

	+  appstream-glib \

We've not needed this for a very long time.

	...
	+  intltool-native \

You don't need this either.

	...
	+  elfutils \

You probably don't want the altos plugin, and can disable this I think.

	+  tpm2-tss \

Is the BMC talking to the TPM? if not, you can drop the TPM plugin and remove this dep.

	+  efibootmgr \

If you're not using UEFI capsules to update then you can drop this too.

	+  protobuf-c \
	+  protobuf-native \
	+  protobuf \

If you're not planning to update logitech bulk controller devices you can drop these too.

**Reply by wltu** [reply]
Thanks for the update. The change I linked to OpenBMC is early WIP and I am not sure what we need yet.

I think you mentioned https://gitlab.com/rossburton/meta-openembedded/-/tree/fwupd/meta-oe/recipes-bsp/fwupd somewhere else. It seems like what they had is actually working. We will work with him to see if that can get submitted to upstream instead.

**Reply by superm1** [reply]
I guess it would be good to understand - what exactly do you want to flash in openbmc? Maybe can drop USB too potentially.

**Reply by hughsie** [reply]
Hint: https://github.com/fwupd/fwupd/pull/3948

**Reply by wltu** [reply]
We wanted to test flashing BIOS image to a SPI flash.. It will be an MTD device.

Oh dang. That's fast. Let me take a look. Thanks
